File: maven/src/main/java/com/antwerkz/critter/CritterContext.java
package com.antwerkz.critter;

import com.google.inject.Singleton;
import org.jboss.forge.roaster.model.source.JavaClassSource;

import java.io.File;
import java.util.ArrayList;
import java.util.Collection;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static java.lang.String.format;

@Singleton
public class CritterContext {
    private final List<String> sourceRoots = new ArrayList<>();
    private Map<String, CritterClass> classes = new HashMap<>();
    private String outputFormat = "java";
    private File outputDirectory;

    public List<String> getSourceRoots() {
        return sourceRoots;
    }

    public void addSourceRoots(List<String> roots) {
        sourceRoots.addAll(roots);
    }

    public void add(CritterClass critterClass) {
        classes.put(format("%s.%s", critterClass.getPackage(), critterClass.getName()), critterClass);
    }

    public CritterClass get(String name) {
        return classes.get(name);
    }

    public Collection<CritterClass> getClasses() {
        return classes.values();
    }

    public boolean isEmbedded(final JavaClassSource clazz) {
        final CritterClass critterClass = get(clazz.getName());
        return critterClass != null && critterClass.isEmbedded();
    }

    public String getOutputFormat() {
        return outputFormat;
    }

    public void setOutputFormat(final String outputFormat) {
        this.outputFormat = outputFormat;
    }

    public void setOutputDirectory(final String directory) {
        this.outputDirectory = new File(directory);
        this.outputDirectory.mkdirs();
    }

    public File getOutputDirectory() {
        return outputDirectory;
    }
}

File: maven/src/main/kotlin/com/antwerkz/critter/CritterContext.kt
package com.antwerkz.critter

import org.jboss.forge.roaster.model.source.JavaClassSource
import java.lang.String.format
import java.util.HashMap

class CritterContext(private val criteriaPkg: String?, val isForce: Boolean) {
    val classes = HashMap<String, CritterClass>()

    fun add(pkgName: String?, critterClass: CritterClass) {
        classes.put(format("%s.%s", pkgName, critterClass.getName()), critterClass)
        if (criteriaPkg != null) {
            critterClass.setPackage(criteriaPkg)
        }
    }

    operator fun get(name: String?): CritterClass? {
        return classes[name]
    }

    fun isEmbedded(name: String?): Boolean {
        return get(name)?.isEmbedded ?: false
    }
}
